_dev/src/components/smart-shopping-campaign/smart-shopping-campaign-table-list.vue

```diff
@@ -318,6 +351,12 @@ export default {
       tableBody.removeEventListener('scroll', this.handleScroll);
     }
   },
+  watch: {
+    typeChosen() {
+      this.fetchCampaigns();
+    },
+  },
```

Comment:
From my understanding, you empty then refill the campaigns from the API?

It would be better to have two different lists, and you load the first page of these two types once this page is loaded. This will allow you to know if you have campaigns the two types. If this is not the case, you will be able to hide the selector, and focus on the list that has results.